contracts/tgrade-valset/src/contract.rs

```diff
@@ -523,6 +529,45 @@ fn list_active_validators(
     })
 }
 
+fn list_jailed_validators(
+    deps: Deps,
+    env: Env,
+    start_after: Option<String>,
+    limit: Option<u32>,
+) -> Result<ListValidatorResponse, ContractError> {
+    let cfg = CONFIG.load(deps.storage)?;
+    let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
+    let start_after = maybe_addr(deps.api, start_after)?;
+    let start = start_after.map(|addr| Bound::exclusive(addr.as_str()));
+
+    let validators: Vec<OperatorResponse> = operators()
+        .range(deps.storage, start, None, Order::Ascending)
+        .map(|r| {
+            let (operator, info) = r?;
+
+            let jailed_until = JAIL
+                .may_load(deps.storage, &Addr::unchecked(&operator))?
+                .filter(|expires| !(cfg.auto_unjail && expires.is_expired(&env.block)));
+            match jailed_until {
+                Some(jailed_into) => Ok(Some(OperatorResponse {
+                    operator: operator.into(),
+                    metadata: info.metadata,
+                    pubkey: info.pubkey.into(),
+                    jailed_until: Some(jailed_into),
+                    active_validator: info.active_validator,
+                })),
+                None => Ok(None),
```

Comment:
Perhaps a `JailedResponse`, with only `operator` and `jailed_until`? That will save the extra load from the storage (pun not intended).

Reply:
Actually `moniker` from metadata is required as well. I guess I could cut `pubkey` and `active_validator`, but would that still save the load?

Reply:
Why is `moniker` required? I would just send the operator address and `jailed_until`. The client can always enrich that if it needs / wants.

Reply:
Consensus is to include that info in response.